app/src/main/java/com/example/tasky/feature_agenda/data/repository/AgendaRemoteRepositoryImpl.kt

```diff
@@ -1,30 +1,65 @@
 package com.example.tasky.feature_agenda.data.repository
 
+import com.example.tasky.agenda_details.domain.model.AgendaItem
 import com.example.tasky.common.data.remote.TaskyApi
 import com.example.tasky.common.domain.Result
 import com.example.tasky.common.domain.error.DataError
 import com.example.tasky.common.domain.util.toNetworkErrorType
 import com.example.tasky.di.AuthenticatedApi
-import com.example.tasky.feature_agenda.domain.model.AgendaResponse
-import com.example.tasky.feature_agenda.domain.model.SyncAgendaResponse
 import com.example.tasky.feature_agenda.domain.repository.AgendaRemoteRepository
 import java.io.IOException
 
 class AgendaRemoteRepositoryImpl(
     @AuthenticatedApi private val api: TaskyApi
 ) : AgendaRemoteRepository {
-    override suspend fun loadAgenda(time: Long): Result<AgendaResponse, DataError.Network> {
+    override suspend fun loadAgenda(time: Long): Result<List<AgendaItem>, DataError.Network> {
         return try {
             val response = api.loadAgenda(time)
             if (response.isSuccessful) {
                 val responseBody = response.body()
-                Result.Success(
-                    AgendaResponse(
-                        events = responseBody?.events ?: emptyList(),
-                        tasks = responseBody?.tasks ?: emptyList(),
-                        reminders = responseBody?.reminders ?: emptyList()
-                    )
-                )
+                if (responseBody != null) {
+                    val agendaItemList: List<AgendaItem> = mutableListOf<AgendaItem>().apply {
```

Comment:
Not a big deal, but you can also quickly combine read-only lists without having to convert them to a mutable one:
```
val list1 = listOf(...)
val list2 = listOf(...)
val list3 = listOf(...)

return list1 + list2 + list3
```

Reply:
good idea! since they are read-only